src/cpp/handler/handlerengine.cpp

```diff
@@ -1789,7 +1790,7 @@ class HandlerEngine::Private : public QObject
 			else if(f.action == PublishFormat::Refresh)
 			{
 				Deferred *d = ControlRequest::refresh(proxyControlClient, i.cid, this);
-				finishedConnection = d->finished.connect(boost::bind(&Private::deferred_finished, this, boost::placeholders::_1, d));
+				finishedConnection[d] = d->finished.connect(boost::bind(&Private::deferred_finished, this, boost::placeholders::_1, d));
```

Comment:
These and others need to be removed from the map when the sender is destroyed.

For the deferreds handled by deferred_finished, it may be enough to have one connections map for all of them.

Reply:
great point! updated 
I have done the same to sessionUpdateMany_finished connections as deferred_finished's